### PR TITLE
refactor: replace formatBalance

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes mf-1714

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-shared/base/src/utils/balance.ts [hunk 1]
```diff
@@ -27,3 +27,9 @@ export function formatBalance(rawValue: BigNumber.Value = '0', decimals = 0, sig
     const raw = negative ? `-${value}` : value
     return raw.includes('.') ? raw.replace(/0+$/, '').replace(/\.$/, '') : raw
 }
+
+export function calcBalance(rawValue: BigNumber.Value = '0', decimals = 0) {
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/utils/balance.ts]: Use `scale10` stead.

File: packages/shared/src/UI/components/FungibleTokenList/FungibleTokenItem.tsx [hunk 2]
```diff
@@ -167,11 +166,7 @@ export const getFungibleTokenItem = <T extends NetworkPluginID>(
             }
             return source !== 'external' || isTrust ? (
                 <Typography className={classes.balance}>
-                    {balance === undefined ? (
-                        <LoadingBase size={24} />
-                    ) : (
-                        Number.parseFloat(new BigNumber(formatBalance(balance ?? 0, decimals, 6)).toFixed(6))
-                    )}
+                    {balance === undefined ? <LoadingBase size={24} /> : scale10(balance ?? 0, decimals).toFixed(6)}
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/FungibleTokenList/FungibleTokenItem.tsx]:
```suggestion
                    {balance === undefined ? <LoadingBase size={24} /> : scale10(balance, decimals).toFixed(6)}
```

File: packages/dashboard/src/pages/Wallets/components/FungibleTokenTableRow/index.tsx [hunk 3]
```diff
@@ -112,8 +103,8 @@ export const FungibleTokenTableRow = memo<TokenTableRowProps>(({ asset, onSend,
             </TableCell>
             <TableCell className={classes.cell} align="center" variant="body">
                 <Typography>
-                    {new BigNumber(formatBalance(asset.balance, asset.decimals)).gt(pow10(-6))
-                        ? Number.parseFloat(toFixed(formatBalance(asset.balance, asset.decimals) ?? '', 6))
+                    {scale10(asset.balance, asset.decimals).gt(pow10(-6))
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/FungibleTokenTableRow/index.tsx]: Cause we don't use `formatBalance` for calculation purposes. We can move this case into `formatBalance`.

File: packages/mask/src/plugins/CryptoartAI/SNSAdaptor/MakeOfferDialog.tsx [hunk 4]
```diff
@@ -142,7 +142,7 @@ export function MakeOfferDialog(props: MakeOfferDialogProps) {
         const balance_ = new BigNumber(balance.value ?? '0')
         if (amount_.isZero()) return t('plugin_collectible_enter_a_price')
         if (amount_.lt(atLeastBidValue)) return t('plugin_collectible_enter_a_price')
-        if (balance_.isZero() || amount_.gt(formatBalance(balance.value, token?.value?.decimals, 4)))
+        if (balance_.isZero() || amount_.gt(scale10(balance.value ?? '0', token?.value?.decimals)))
```
Comment (guanbinrui) [inline packages/mask/src/plugins/CryptoartAI/SNSAdaptor/MakeOfferDialog.tsx]: They are different.

File: packages/mask/src/plugins/ITO/SNSAdaptor/ConfirmDialog.tsx [hunk 5]
```diff
@@ -39,7 +39,7 @@ function SwapItem(props: SwapItemProps) {
     const { classes } = useSwapItemStyles()
     const { t } = useI18N()
 
-    const amount_ = formatBalance(swapAmount || '0', swap?.decimals)
+    const amount_ = scale10(swapAmount || '0', swap?.decimals)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/ITO/SNSAdaptor/ConfirmDialog.tsx]: Caution! They return different results.

File: packages/dashboard/src/pages/Wallets/components/FungibleTokenTableRow/index.tsx [hunk 6]
```diff
@@ -112,8 +111,8 @@ export const FungibleTokenTableRow = memo<TokenTableRowProps>(({ asset, onSend,
             </TableCell>
             <TableCell className={classes.cell} align="center" variant="body">
                 <Typography>
-                    {new BigNumber(formatBalance(asset.balance, asset.decimals)).gt(pow10(-6))
-                        ? Number.parseFloat(toFixed(formatBalance(asset.balance, asset.decimals) ?? '', 6))
+                    {leftShift(asset.balance, asset.decimals).gt(pow10(-6))
+                        ? formatBalance(asset.balance, asset.decimals, 6)
```
Comment (guanbinrui) [inline packages/dashboard/src/pages/Wallets/components/FungibleTokenTableRow/index.tsx]: Move this case into formatBalance.

**Review by guanbinrui — APPROVED** [review]

**Review by UncleBill — APPROVED** [review]

**Pull request merged**